pkg/report/report.go

```diff
@@ -21,7 +23,36 @@ type ScanReportParser interface {
 	Parse(string) (*types.UpdateManifest, error)
 }
 
-func TryParseScanReport(file string) (*types.UpdateManifest, error) {
+func TryParseScanReport(file, scanner string) (*types.UpdateManifest, error) {
```

Comment:
can you add a test provider (can use existing trivy code) and make sure to add tests

Reply:
just to confirm, I should add a provider file like https://github.com/open-policy-agent/gatekeeper/blob/master/test/externaldata/dummy-provider/provider.go right? sorry if this is a silly question as I'm new to test providers

Reply:
no worries! we don't need the same or similar code (just the concept). if we can turn trivy into a test provider and without duplicating code, i think that would be ideal so we can make sure to test the provider call flow.

Reply:
can you please check my new commits regarding tests, is it what is expected?

Reply:
yes! this is great for now. we can do a follow up to integrate https://github.com/project-copacetic/scanner-parser-template/ to test success cases into e2e

as a follow up, we'll need:
- go install https://github.com/project-copacetic/scanner-parser-template in github actions e2e
- run with a fake report
- make sure copa can call and receive results from the plugin

Reply:
we will need to release a version of the scanner-parser-template if we want to install it for testing, right?
For example: In the file `.github/workflows/scripts/download-tooling.sh` we can install it using something like this:
```bash
curl -sfL https://github.com/project-copacetic/scanner-parser-template/releases/download/v${SCANNER_PARSER_TEMPLATE_VERSION}/scanner-parser-template_${SCANNER_PARSER_TEMPLATE_VERSION}_Linux-64bit.tar.gz -o scanner-parser-template.tar.gz
sudo tar -zxvf scanner-parser-template.tar.gz -C /usr/local/bin/
rm scanner-parser-template.tar.gz

Reply:
@sozercan in order to go install, we will need a release of the scanner-plugin-template, right?
We can also install using something like:
```bash
curl -sfL https://github.com/project-copacetic/scanner-plugin-template/releases/download/v${SCANNER_PLUGIN_TEMPLATE_VERSION}/scanner-plugin-template_${SCANNER_PLUGIN_TEMPLATE_VERSION}_Linux-64bit.tar.gz -o scanner-plugin-template.tar.gz
sudo tar -zxvf scanner-plugin-template.tar.gz -C /usr/local/bin/
rm scanner-plugin-template.tar.gz
```
similar to how [other tools are installed](https://github.com/project-copacetic/copacetic/blob/main/.github/workflows/scripts/download-tooling.sh) in the copa CI, however even that would require a release of the fake-plugin.